src/screens/Settings/hooks/useSettings.js

```diff
@@ -33,6 +33,7 @@ const useSettings = () => {
   const logout = async () => {
     try {
       await OneSignalUtil.removeAllSubscribedTags();
+      OneSignalUtil.removeExternalId();
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The addition of `OneSignalUtil.removeExternalId();` is a good step towards ensuring that the user's external ID is removed from OneSignal when they log out. However, it's important to handle potential errors that might occur during this operation. Consider wrapping this line in a try-catch block to handle any exceptions that might be thrown.
```diff
-       OneSignalUtil.removeExternalId();
+       try {
+           await OneSignalUtil.removeExternalId();
+       } catch (error) {
+           console.error('Failed to remove external ID:', error);
+       }
```
This way, even if removing the external ID fails for some reason, the logout process can continue and the error can be logged for debugging purposes.


<!-- This is an auto-generated comment by OSS CodeRabbit -->

Reply:
no need, after logged out, logging in will reset the external id anyway